### ISSUE TITLE
[AVT] autoFocus is not accepted by Dropdown component

### ISSUE THREAD

**Issue by yanweibj** [carbon-design-system/carbon#5401]
> Operating system
MacOS
> Browser
Chrome
> Automated testing tool and ruleset
DAP 
IBM Accessibility January 2020
IBM Accessibility Experimental

## Detailed description

> What version of the Carbon Design System are you using?
Carbon v10
> What did you expect to happen?
autoFocus should be applied to Dropdown component
> What happened instead?
If I apply autoFocus prop to TextInput it works fine but applying autoFocus to Dropdown component it does not work.

> What WCAG 2.1 checkpoint does the issue violate?
2.4.3 Focus Order

## Steps to reproduce the issue

1.  Have a form on page with Dropdown on it together with other component
2.  Apply autoFocus to Dropdown component
3.  See that Dropdown is not auto focused

**Comment by emyarod** [comment]
autofocus is exclusive to `input` and `select` elements I believe, so it would not be supported by our dropdown

**Comment by yanweibj** [comment]
<img width="625" alt="image" src="https://user-images.githubusercontent.com/16932498/75252562-eb190900-5817-11ea-838d-a8771f088849.png">

**Comment by joshblack** [comment]
Going to close this out as I believe autofocus is only available on form and `button` elements as @emyarod noted above. Unfortunately, this isn't a behavior that we could forward along to the dropdown implementation which does not use any of these types of elements under the hood :disappointed:

Let us know if there is something we're missing here!
